src/librustc_resolve/lib.rs

```diff
@@ -3512,6 +3668,41 @@ fn path_names_to_string(path: &Path, depth: usize) -> String {
     names_to_string(&names[..])
 }
 
+fn candidates_to_suggestion(paths: &Vec<Path>) -> Vec<String> {
+
+    let mut suggestion_vec = Vec::new();
+    let mut path_strings: Vec<_> = paths.into_iter()
+                                        .map(|p| path_names_to_string(&p, 0))
+                                        .collect();
+    // we want consistent results across executions, but candidates are produced
+    // by iterating through a hash map, so make sure they are ordered:
+    path_strings.sort();
+
+    if !paths.is_empty() {
+        if paths.len() == 1 {
+            let mut suggestion_str = String::new();
+            suggestion_str.push_str("You may be referring to `");
+            suggestion_str.push_str(&path_strings[0]);
+            suggestion_str.push_str("`, which you need to import with the `use` keyword.");
+            suggestion_vec.push(suggestion_str);
+        } else {
+            let mut suggestion_str = String::new();
+            suggestion_str.push_str("You may be referring to one of: ");
+            suggestion_vec.push(suggestion_str);
+            for path in path_strings {
+                let mut proposal_str = String::new();
+                proposal_str.push_str(" * `");
```

Comment:
I think drop the `*` and just add two spaces in front of the string; this is what we do in `librustc/middle/traits/error_reporting.rs` in a similar scenario (suggesting impls)